### PR TITLE
Support for a remote db config

### PR DESCRIPTION
**Pull request by ttung**
### Description
For the most part, accessing the DB is a function of AWS_PROFILE or AWS_ACCESS_KEY_ID/AWS_SECRET_KEY_ID.  We want to have an object that encompasses all the remote db config so we can access the DB from the CLI.

If we didn't do this, then we would require the user to specify --staging or --prod (to initialize `StagingConfig` or `ProductionConfig`), when it actually doesn't matter.

### Test plan
flask run

### CODE REVIEW

File: src/py/aspen/config/config.py [hunk 1]
```diff
@@ -58,26 +58,29 @@ class Config:
...
-            raise Exception(
-                f"Attributes ({missing_properties_string}) defined as flask properties"
-                f" in Config but not in {cls.__name__}"
-            )
+        if descriptive_name is not None:
+            Config._subclasses[descriptive_name] = cls
+
+            # get a list of @flaskproperties for Config and the subclass of Config, and
+            # compare to see if there are any attributes that were marked as
+            # @flaskproperty in Config but not in the subclass.
```
Comment (phoenixAja) [inline src/py/aspen/config/config.py:72]: don't ABC's do something similar to this?

File: src/py/aspen/config/config.py [hunk 2]
```diff
@@ -187,3 +189,11 @@ def DATABASE_URI(self) -> str:
     @property
     def DATABASE_READONLY_URI(self) -> str:
         raise NotImplementedError()
+
+
+class RemoteConfig(Config):
```
Comment (phoenixAja) [inline src/py/aspen/config/config.py]: i feel like a more intuitive name might be `RemoteDBConfig`? Unless we're planning on using this config for things other that a DB down the line

File: src/py/aspen/config/production.py [hunk 3]
```diff
@@ -1,12 +1,10 @@
-import uuid
-
 from aspen.config import config
 
 
-class ProductionConfig(config.Config, descriptive_name="prod"):
+class ProductionConfig(config.RemoteConfig, descriptive_name="prod"):
     @config.flaskproperty
-    def SECRET_KEY(self) -> str:
-        return uuid.uuid4().hex
```
Comment (phoenixAja) [inline src/py/aspen/config/production.py:9]: do we need to add `SECRET_KEY` to the `RemoteConfig` if we're removing it here?

File: src/py/aspen/config/staging.py [hunk 4]
```diff
@@ -1,29 +1,20 @@
...
-class StagingConfig(config.Config, descriptive_name="staging"):
+class StagingConfig(config.RemoteConfig, descriptive_name="staging"):
     @config.flaskproperty
     def DEBUG(self) -> bool:
         return True
 
-    @config.flaskproperty
-    def SECRET_KEY(self) -> str:
-        return uuid.uuid4().hex
-
```
Comment (phoenixAja) [inline src/py/aspen/config/staging.py:18]: samesies

File: src/py/aspen/config/production.py (hunk 3 above)
Comment (phoenixAja) [reply src/py/aspen/config/production.py:9]: ok i see you added this back in later PRs, nevermind

**Review by phoenixAja — APPROVED** [review]

File: src/py/aspen/config/config.py (hunk 1 above)
Comment (ttung) [reply src/py/aspen/config/config.py:72]: If you tag something as an abstract class (abc.abstractclass), or there are abstract methods (abc.abstractmethod), it cannot be instantiated.  This is similar in concept but we can't quite use those constructs to achieve the same.

File: src/py/aspen/config/config.py (hunk 2 above)
Comment (ttung) [reply src/py/aspen/config/config.py]: It's subclassed by all configs that use a remote db, so I guess you're right.

File: src/py/aspen/config/production.py (hunk 3 above)
Comment (ttung) [reply src/py/aspen/config/production.py:9]: A terrible implementation is provided by the base class now.

File: src/py/aspen/config/config.py (hunk 2 above)
Comment (ttung) [reply src/py/aspen/config/config.py]: Updated.

**Pull request merged**